ai-server/main.py

```
@@ -59,53 +79,114 @@ def health():
+
+
+def _normalize_text(text: str) -> str:
+    """
+    opendataloader 아티팩트 제거: 한글 문장 중간에 삽입되는 과도한 개행을 정규화한다.
+    Docling 출력에도 동일하게 적용해도 무해하다.
+    """
+    # 한글 사이 3개 이상 개행 제거 (파서 버그로 삽입되는 아티팩트)
+    text = re.sub(r'([가-힣])\n{3,}([가-힣])', r'\1\2', text)
+    # 페이지 경계의 과도한 개행 정규화 (삼중 이상 → 이중)
+    text = re.sub(r'\n{3,}', '\n\n', text)
+    return text
```

Ruff (0.15.10)

[warning] Unnecessary assignment to `text` before `return` statement

Remove unnecessary assignment

(RET504)